streams/src/test/java/org/apache/kafka/streams/integration/MetricsIntegrationTest.java

```diff
@@ -377,7 +378,7 @@ private void shouldAddMetricsOnAllLevels(final String builtInMetricsVersion) thr
             builtInMetricsVersion
         );
         checkCacheMetrics(builtInMetricsVersion);
-
+        verifyFailedStreamThreadsSensor(0.0);
```

Comment:
We also need to verify that the metric works when there is a failed stream thread. Options are (1) to create a custom processor now and (IIRC) run the test suite twice, once with failing stream threads and once without to confirm that the metric works. I'm not sure if the custom processor will let us just fail one stream thread right before closing the app. Or (2) wait until add/remove stream threads is implemented and remove threads and test the metric after removing some threads before closing the app. WDYT?

Reply:
I think you should try using the custom processor. You can find an example in `StreamsUncaughtExceptionHandlerIntegrationTest.java`

Reply:
I would put the test whether the metric is recorded correctly in `StreamThreadTest`. An example for such a test is `shouldLogAndRecordSkippedRecordsForInvalidTimestamps()`. I do not think an integration test is needed. The test regarding the existence of the metric, i.e., `checkMetricByName(listMetricThread, FAILED_STREAM_THREADS, 1);` should stay here.

Reply:
After looking at both test classes, I think it actually might make the most sense to put the test for this metric in `StreamsUncaughtExceptionHandlerIntegrationTest`, since the metric is so closely aligned with the exception handler anyways and the setup works nicely with what we're trying to test with the metric. From the size + complexity of the other test classes, I think creating an overloaded processor for one test out of 20+ tests seems tricky.